app/src/pages/inside/logsPage/logsGridToolbar/logsGridToolbar.jsx

```diff
@@ -0,0 +1,176 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import classNames from 'classnames/bind';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
+import Parser from 'html-react-parser';
+import { userIdSelector } from 'controllers/user';
+import { InputSlider } from 'components/inputs/inputSlider';
+import { InputCheckbox } from 'components/inputs/inputCheckbox';
+import { GhostButton } from 'components/buttons/ghostButton';
+import { Pagination } from './pagination';
+import { getLogLevel, LOG_LEVELS, setLogLevel } from './utils/logLevel';
+import { getWithAttachments, setWithAttachments } from './utils/withAttachments';
+import ConsoleIcon from './img/console-inline.svg';
+import MarkdownIcon from './img/markdown-inline.svg';
+import styles from './logsGridToolbar.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  withAttachments: {
+    id: 'LogsGridToolbar.withAttachments',
+    defaultMessage: 'Logs with Attachment',
+  },
+  markdownMode: {
+    id: 'LogsGridToolbar.markdownMode',
+    defaultMessage: 'Markdown Mode',
+  },
+  consoleView: {
+    id: 'LogsGridToolbar.consoleView',
+    defaultMessage: 'Console View',
+  },
+  nextError: {
+    id: 'LogsGridToolbar.nextError',
+    defaultMessage: 'Next Error',
+  },
+});
+
+@injectIntl
+@connect((state) => ({
+  userId: userIdSelector(state),
+}))
+export class LogsGridToolbar extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    activePage: PropTypes.number.isRequired,
+    pageCount: PropTypes.number.isRequired,
+    onChangePage: PropTypes.func.isRequired,
+    userId: PropTypes.string.isRequired,
+    initialLogLevel: PropTypes.string,
+    children: PropTypes.func,
+    onChangeLogLevel: PropTypes.func,
+    onChangeWithAttachments: PropTypes.func,
+  };
+
+  static defaultProps = {
+    initialLogLevel: null,
+    children: () => {},
+    onChangeLogLevel: () => {},
+    onChangeWithAttachments: () => {},
+  };
+
+  state = {
+    markdownMode: false,
+    consoleView: false,
+    withAttachments: getWithAttachments(this.props.userId),
+    logLevel: getLogLevel(this.props.initialLogLevel, this.props.userId),
```

Comment:
I think that we shouldn't copy prop `initialLogLevel` to the state. The case: if we change URL, this change won't be reflected on the page.
Let's make URL the main source of truth here. You can also cache default value to reduce localStorage usage on every render.

Reply:
Yes you right. Fixed.